components/scream/cmake/ctest_script.cmake

```diff
@@ -45,10 +45,19 @@ else ()
       endif()
   else()
     if (NOT BUILD_ONLY)
+
+      if (DEFINED ENV{SCREAM_FORCE_RUN_DIFF})
+        set(INCLUDE_REGEX "p3_run_and_cmp")
+      elseif (DEFINED ENV{SCREAM_FORCE_RUN_DIFF_BFB_UNIT})
+        set(INCLUDE_REGEX "p3_tests_ut")
```

Comment:
I'm not following here. Is the SCREAM_FORCE_RUN_DIFF option only forcing a diff in baselines tests, while SCREAM_FORCE_RUN_DIFF_BFB_UNIT forces a diffs in the unit tests only? Why isn't `RUN_DIFF` including also unit tests? Maybe I'm confused by the var names... And why is this logic applied only to P3?

Reply:
On a side note, perhaps a better way to handle this would be through labels? A regex would have to be modified every time we have a new test that needs to be added, while using specific labels would not require the ctest script to be updated every time.

Reply:
@bartgol , yes, we want SCREAM_FORCE_RUN_DIFF to only run the p3_run_and_cmp test. If we ran all the tests, the BFB ones would fail too since we "sabotage" rain_sed in order to get a diff. This logic applies to p3 because we leverage p3 to test diffs. In the main CMakeLists.txt file:

```
if ($ENV{SCREAM_FORCE_RUN_DIFF})
  add_definitions(-DSCREAM_FORCE_RUN_DIFF)
  set(SCREAM_FAKE_NEED_P3 ON)
endif()
if ($ENV{SCREAM_FORCE_RUN_DIFF_BFB_UNIT})
  add_definitions(-DSCREAM_FORCE_RUN_DIFF_BFB_UNIT)
  set(SCREAM_FAKE_NEED_P3 ON)
endif()
```

So, if SCREAM_FAKE_ONLY is on, we build almost none of SCREAM, but then we don't have enough to test that diffs are detected, so we add p3 if it looks like that's what we are trying to test.

Similar logic would be needed with labels. `INCLUDE_REGEX` is used to limit what tests are run by ctest so, if we are trying to detect a baseline diff, we only run the baseline test (p3_run_and_cmp). I hope that makes sense.

Reply:
Ok, regex or label would not change much, so that's fine.